model/task/task.go

```diff
@@ -1109,10 +1111,24 @@ func (t *Task) MarkAsContainerDeallocated(ctx context.Context, env evergreen.Env
 	}
 
 	t.ContainerAllocated = false
-	t.DispatchTime = utility.ZeroTime
```

Comment:
What's the reason we don't need to set these props on the task anymore?

Reply:
I originally modeled some of this logic based on [MarkAsHostUndispatched](https://github.com/evergreen-ci/evergreen/blob/5cd7c47bca7225acdc715a514f67a03a911b3621/model/task/task.go#L1028-L1051) but changed my mind about the necessity of some of the fields which are specific to dispatching the task. Allocating/deallocating a container is not really equivalent to dispatching/undispatching a task because dispatch is the moment that the task is assigned to run on a host/container, whereas container allocation is when the task is given a container that will eventually run it. Since they're separate parts of the task lifecycle, it doesn't make sense to unset them here.